lapce-proxy/src/plugin.rs

```diff
@@ -183,6 +187,65 @@ impl PluginCatalog {
         Ok(plugin)
     }
 
+    fn get_plugin_env(
+        &mut self,
+        plugin_desc: &PluginDescription,
+    ) -> Vec<(String, String)> {
+        let mut vars = Vec::new();
+
+        let conf = match plugin_desc.configuration.clone() {
+            Some(val) => val,
+            None => return vars,
+        };
+
+        let t = match serde_json::from_value::<PluginConfiguration>(conf) {
+            Ok(val) => val,
+            Err(_err) => {
+                // We do not print any error as the primary error is missing field, which is allowed.
+                // Note that the error is a custom std::Error from serde:
+                // serde::de::Error::missing_field("missing field `env_command`") and as such cannot be checked
+                // unless we create one to compare with.
+                return vars;
+            }
+        };
+
+        let args = t.env_command.as_str();
+
+        let output = if cfg!(target_os = "windows") {
+            Command::new("cmd").arg("/c").arg(args).output()
+        } else {
+            Command::new("sh").arg("-c").arg(args).output()
+        };
+
+        let output = match output {
+            Ok(val) => val,
+            Err(err) => {
+                println!(
+                    "Error during env command execution for plugin {}: {}",
+                    plugin_desc.name, err
+                );
+                return vars;
+            }
+        };
+
+        let data = match String::from_utf8(output.stdout) {
+            Ok(val) => val,
+            Err(_err) => String::from(""),
+        };
+
+        let lines = data.lines();
+
+        for l in lines {
+            let s: Vec<&str> = l.split('=').collect();
+            if s.len() < 2 {
+                continue;
+            }
+            vars.push((String::from(s[0]), String::from(s[1])));
```

Comment:
```suggestion
            let mut s = l.split('=');
            let key = s.next().unwrap();
            let val = if let Some(val) = s.next() {
                val
            } else {
                continue;
            };            
            vars.push((String::from(key), String::from(val)));
```

Or, since `a=b=c` might be invalid anyway, [`str::split_once`](https://doc.rust-lang.org/std/primitive.str.html#method.split_once) might be better here. That way we don't ignore anything but might pass a "wrong" value.